### PR TITLE
Support Feature Flags for Breaking Changes

### PR DESCRIPTION
**Pull request by demirkayaender**
**What changed?**
Note: ignore go.mod changes for now. They will be fixed once the client changes are in

Shipping WorkflowExecutionAlreadyCompletedError broke some customers' workflow logic since it was changing previous contract between their workflow and cadence server. With this change, they can enable the new feature only when they are ready. 

Changes need to go in with the following orders
- IDL changes: https://github.com/uber/cadence-idl/pull/73
- Client-side changes: https://github.com/uber-go/cadence-client/pull/1103
- Server-side changes: https://github.com/uber/cadence/pull/4257
- Samples changes: https://github.com/uber-common/cadence-samples/pull/45

Ignore go.mod/sum changes for now, I will update them once IDL and client changes are in. 

**Why?**
Without this change, shipping new features will break customer workflows.

**How did you test it?**
`go test -run TestIntegrationSuite github.com/uber/cadence/host -testify.m TestSignalWorkflow$ -v`

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -83,6 +83,8 @@ require (
 	honnef.co/go/tools v0.0.1-2020.1.4 // indirect
 )
 
+replace go.uber.org/cadence => /Users/enderd/cadence-client
```
Comment (yycptt) [inline go.mod:86]: Revert this change? I think it's only for local test?

**Review by emrahs — APPROVED** [review]

**Comment by JuanPalacios** [comment]
Is this feature included in the java client?

**Comment by demirkayaender** [comment]
Yes, java client supports it. Here's an example usage: https://github.com/demirkayaender/cadence-java-samples/blob/e3a0e310435d51c7152b2c904eef960d22fa4f5a/src/main/java/com/uber/cadence/samples/hello/HelloSignal.java#L103
